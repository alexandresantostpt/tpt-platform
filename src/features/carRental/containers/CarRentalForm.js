import { connect } from 'react-redux'

import { getObjSelector as travelGetObjSelector, getClientsSelector } from '@features/travelScript/selectors'

import { requestDelAction, requestGetSingleAction, requestSaveDataAction } from '../actions'

import { getObjSelector } from '../selectors'

import CarRentalForm from '../pages/CarRentalForm'

const mapStateToProps = state => ({
    clients: getClientsSelector(state),
    rent: getObjSelector(state),
    travelScript: travelGetObjSelector(state)
})

const mapDispatchToProps = {
    requestDelAction,
    requestGetSingleAction,
    requestSaveDataAction
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(CarRentalForm)
