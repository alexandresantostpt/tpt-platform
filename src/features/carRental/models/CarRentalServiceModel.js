import { config } from '@config'

import moment from 'moment'

import { normalizeTimezone } from '@utils/date'

import CarRentalLibraryModel from './CarRentalLibraryModel'

class CarRentalServiceModel {
    constructor(
        {
            __v,
            _id,
            addressDevolution,
            addressRetire,
            agency,
            carModel,
            createdAt,
            devolution,
            devolutionDate,
            devolutionName,
            hourDevolution,
            hourRetire,
            library,
            operationHourEnd,
            operationHourStart,
            passengerResponsible,
            phoneNumberDevolution,
            phoneNumberRetire,
            requiredDocuments,
            retire,
            retireDate,
            retireName,
            scriptDate,
            siteRetire,
            travel,
            updatedAt
        },
        fromForm = false
    ) {
        this.__v = __v
        this._id = _id
        this._agency = agency
        this._addressDevolution = addressDevolution
        this._addressRetire = addressRetire
        this._carModel = carModel
        this._createdAt = moment(createdAt).utc(false)
        this._devolution = devolution
        this._devolutionDate = moment(devolutionDate).utc(false)
        this._devolutionName = devolutionName
        this._hourDevolution = fromForm ? normalizeTimezone(hourDevolution) : moment(devolution && devolution.hour).utc(false)
        this._hourRetire = fromForm ? normalizeTimezone(hourRetire) : moment(retire && retire.hour).utc(false)
        this._library = library ? new CarRentalLibraryModel(library) : null
        this._operationHourEnd = moment(operationHourEnd).utc(false)
        this._operationHourStart = moment(operationHourStart).utc(false)
        this._passengerResponsible = passengerResponsible
        this._phoneNumberDevolution = phoneNumberDevolution
        this._phoneNumberRetire = phoneNumberRetire
        this._requiredDocuments = requiredDocuments
        this._retire = retire
        this._retireDate = moment(retireDate).utc(false)
        this._retireName = retireName
        this._scriptDate = scriptDate
        this._siteRetire = siteRetire
        this._travel = travel
        this._updatedAt = moment(updatedAt).utc(false)
    }

    get v() {
        return this.__v
    }

    get id() {
        return this._id
    }

    get agency() {
        return this._agency
    }

    get addressDevolution() {
        return this._addressDevolution
    }

    get addressRetire() {
        return this._addressRetire
    }

    get createdAt() {
        return this._createdAt
    }

    get devolutionDate() {
        return this._devolutionDate
    }

    get carModel() {
        return this._carModel
    }

    get devolution() {
        return this._devolution
    }

    get devolutionName() {
        return this._devolutionName
    }

    get hourDevolution() {
        return this._hourDevolution
    }

    get hourRetire() {
        return this._hourRetire
    }

    get library() {
        return this._library
    }

    get operationHourEnd() {
        return this._operationHourEnd
    }

    get operationHourStart() {
        return this._operationHourStart
    }

    get passengerResponsible() {
        return this._passengerResponsible
    }

    get phoneNumberDevolution() {
        return this._phoneNumberDevolution
    }

    get phoneNumberRetire() {
        return this._phoneNumberRetire
    }

    get requiredDocuments() {
        return this._requiredDocuments
    }

    get retire() {
        return this._retire
    }

    get retireDate() {
        return this._retireDate
    }

    get retireName() {
        return this._retireName
    }

    get scriptDate() {
        return this._scriptDate
    }

    get siteRetire() {
        return this._siteRetire
    }

    get updatedAt() {
        return this._updatedAt
    }

    getImage() {
        return this.library && this.library.image && `${config.api.url}/car-rental/download/${this.library.image}`
    }

    toForm() {
        return {
            __v: this.v,
            _id: this.id,
            addressDevolution: this.devolution.address,
            addressRetire: this.retire.address,
            carModel: this.carModel,
            devolutionDate: this.devolution.date,
            devolutionName: this.devolution.name,
            fees: this.fees,
            hourDevolution: this.hourDevolution,
            hourRetire: this.hourRetire,
            library: this.library ? this.library.toForm() : null,
            passengerResponsible: this.passengerResponsible,
            phoneNumberDevolution: this.devolution.phoneNumber,
            phoneNumberRetire: this.retire.phoneNumber,
            requiredDocuments: this.requiredDocuments.join(''),
            retireDate: this.retire.date,
            retireName: this.retire.name,
            scriptDate: this.scriptDate,
            siteRetire: this.retire.site
        }
    }

    /* eslint-disable-next-line complexity */
    toDTO() {
        return {
            __v: this.v,
            _id: this.id,
            carModel: this.carModel || null,
            devolution: {
                address: this.addressDevolution || null,
                date: this.devolutionDate || null,
                hour: this.hourDevolution || null,
                name: this.devolutionName || null,
                operationHourEnd: '2019-02-01T18:00:00.000Z',
                operationHourStart: '2019-02-01T09:00:00.000Z',
                phoneNumber: this.phoneNumberDevolution || null
            },
            library: this.library ? this.library.toDTO() : null,
            passengerResponsible: this.passengerResponsible || null,
            requiredDocuments: this.requiredDocuments || null,
            retire: {
                address: this.addressRetire || null,
                date: this.retireDate || null,
                hour: this.hourRetire || null,
                name: this.retireName || null,
                operationHourEnd: '2019-02-01T18:00:00.000Z',
                operationHourStart: '2019-02-01T09:00:00.000Z',
                phoneNumber: this.phoneNumberRetire || null,
                site: this.siteRetire || null
            },
            scriptDate: this.scriptDate
        }
    }
}

export default CarRentalServiceModel
