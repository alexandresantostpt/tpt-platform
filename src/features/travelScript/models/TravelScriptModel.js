import moment from 'moment'

import Travel from '../../travel/TravelModel'
import TravelDateModel from './TravelDateModel'
import { isNotEmpty } from '@/utils/functions'

class TravelScriptModel {
    constructor({ __v, _id, dates, deleted, edited, travel, createdAt, updatedAt }) {
        this.__v = __v
        this._id = _id
        this._dates = dates && [...dates].map(date => new TravelDateModel(date))
        this._deleted = deleted
        this._edited = edited
        this._travel = new Travel(travel)
        this._createdAt = moment(createdAt).utc(false)
        this._updatedAt = moment(updatedAt).utc(false)
    }

    get v() {
        return this.__v
    }

    get id() {
        return this._id
    }

    get dates() {
        return this._dates
    }

    get deleted() {
        return this._deleted
    }

    get edited() {
        return this._edited
    }

    get travel() {
        return this._travel
    }

    get createdAt() {
        return this._createdAt
    }

    get updatedAt() {
        return this._updatedAt
    }

    hasServices() {
        return this.dates.some(date => date.services.length)
    }

    getRangeDates() {
        if (isNotEmpty(this._travel)) {
            return { endDate: this._travel.travelEndDate, startDate: this._travel.travelStartDate }
        }

        return {}
    }

    getClientsId() {
        return [...this.travel.clients].map(({ _id }) => _id)
    }

    toForm() {
        return {
            dates: this.dates,
            edited: this.edited,
            id: this.id,
            travel: this.travel,
            v: this.v
        }
    }
}

export default TravelScriptModel
