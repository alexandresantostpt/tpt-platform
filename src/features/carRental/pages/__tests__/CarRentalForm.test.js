import React from 'react'
import renderer from 'react-test-renderer'

import CarRentalServiceModel from '../../models/CarRentalServiceModel'

import CarRentalForm from '../CarRentalForm'
import TravelScriptModel from '@/features/travelScript/models/TravelScriptModel'

describe('Tests for CarRentalForm component', () => {
    const rent = new CarRentalServiceModel({
        __v: 1,
        _id: '5c815c95650a39608fd4923f',
        carModel: 'modelo do carro',
        createdAt: '2019-03-07T18:01:57.211Z',
        devolution: {
            address: 'endereço de devolução',
            date: '2019-03-14T15:00:00.000Z',
            hour: '2019-03-07T22:00:00.434Z',
            name: 'Nome do local de devolução',
            operationHourEnd: '2019-02-01T18:00:00.000Z',
            operationHourStart: '2019-02-01T09:00:00.000Z',
            phoneNumber: 'telefone da locadora'
        },
        isIncluded: 'teste',
        name: 'Nome da locadora',
        passengerResponsible: ['5c757fb2f1a4a01bbfe7b21b'],
        requiredDocuments: ['cnh'],
        retire: {
            address: 'Endereço de retirada',
            date: '2019-03-07T15:00:00.000Z',
            hour: '2019-03-07T20:00:00.463Z',
            name: 'Nome local de retirada',
            operationHourEnd: '2019-02-01T18:00:00.000Z',
            operationHourStart: '2019-02-01T09:00:00.000Z',
            phoneNumber: 'Telefone da locadora'
        },
        script: '5c757fb2f1a4a01bbfe7b21d',
        scriptDate: '2019-02-03T00:00:00.000Z',
        type: 'CAR_RENTAL',
        updatedAt: '2019-03-07T18:02:27.854Z'
    })
    const travelScript = new TravelScriptModel({ travel: { __v: 0 } })

    it('Should render withou crash', () => {
        const match = {
            params: {
                id: '5c52ebcd5fc7c949223ab705'
            }
        }
        const tree = renderer
            .create(
                <CarRentalForm
                    match={match}
                    rent={rent}
                    requestDelAction={jest.fn()}
                    requestGetSingleAction={jest.fn()}
                    requestSearchAction={jest.fn()}
                    travelScript={travelScript}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
