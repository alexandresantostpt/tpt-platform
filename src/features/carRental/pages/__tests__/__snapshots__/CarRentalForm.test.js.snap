// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CarRentalForm component Should render withou crash 1`] = `
Array [
  <div
    className="WrapperPadding-ovjbyx-0 lcnoja"
  >
    <div
      className="Panel__StyledPanel-sc-1qzkpb6-0 cGUlCn"
    >
      <span
        className="Panel__StyledIcon-sc-1qzkpb6-2 iavvSk"
      >
        <span
          className="icon icon-car Icon__StyledIcon-crr1cg-0 dfhcZR"
          disabled={false}
          fontSize="3rem"
          onClick={null}
        />
      </span>
      <div
        className="UploadButton__StyledContainer-sc-1p8zoc-0 fpZFIG"
      >
        <label
          className="UploadButton__StyledUploadButton-sc-1p8zoc-3 ivhrYW"
          htmlFor="inputFile"
        >
          +
        </label>
        <input
          accept="image/png, image/jpeg"
          className="UploadButton__InputFile-sc-1p8zoc-4 eLaooA"
          id="inputFile"
          onChange={[Function]}
          type="file"
        />
      </div>
    </div>
    <ul
      className="PanelActions__StyledActions-sc-8xk8c8-0 lllkGw"
    />
  </div>,
  <form
    className="FormBuilder__StyledForm-c1htou-0 PZxQa"
    onSubmit={[Function]}
  >
    <div
      className="form__container"
    >
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-id"
          disabled={false}
          id="id"
          maxLength={255}
          name="id"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="hidden"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-v"
          disabled={false}
          id="v"
          maxLength={255}
          name="v"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="hidden"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Nome da locadora
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-library.name"
          disabled={false}
          id="name"
          maxLength={120}
          name="library.name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Nome do local de retirada
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-retireName"
          disabled={false}
          id="retireName"
          maxLength={120}
          name="retireName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Nome local de retirada"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Endereço de retirada
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-addressRetire"
          disabled={false}
          id="addressRetire"
          maxLength={255}
          name="addressRetire"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Endereço de retirada"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Telefone da locadora
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-phoneNumberRetire"
          disabled={false}
          id="phoneNumberRetire"
          maxLength={30}
          name="phoneNumberRetire"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Telefone da locadora"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Site da locadora
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-siteRetire"
          disabled={false}
          id="siteRetire"
          maxLength={255}
          name="siteRetire"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Data de retirada
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormDate__StyledFormDate-sc-1cgz0rk-0 ebKpCI"
        >
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-"
                    aria-label="Selecione uma data..."
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                    disabled={false}
                    id=""
                    name=""
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Selecione uma data..."
                    readOnly={true}
                    required={false}
                    type="text"
                    value="07 Mar 2019"
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <button
                  className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon icon-calendar Icon__StyledIcon-crr1cg-0 cmjVJQ"
                    disabled={false}
                    fontSize="1.75rem"
                    onClick={null}
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Horário de retirada
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormTime__StyledFormTime-sc-1ka3ing-1 bdjJPj"
        >
          <input
            className="FormTime__StyledDatePicker-sc-1ka3ing-0 hCpkqS"
            disabled={false}
            onChange={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  2,
                ],
                "delimiters": Array [
                  ":",
                ],
              }
            }
            placeholderText="Selecione um horário..."
            value="17:00"
          />
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Nome do local de devolução
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-devolutionName"
          disabled={false}
          id="devolutionName"
          maxLength={120}
          name="devolutionName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Nome do local de devolução"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Endereço de devolução
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-addressDevolution"
          disabled={false}
          id="addressDevolution"
          maxLength={255}
          name="addressDevolution"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="endereço de devolução"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Telefone de devolução
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-phoneNumberDevolution"
          disabled={false}
          id="phoneNumberDevolution"
          maxLength={30}
          name="phoneNumberDevolution"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value="telefone da locadora"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Data de devolução
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormDate__StyledFormDate-sc-1cgz0rk-0 ebKpCI"
        >
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-"
                    aria-label="Selecione uma data..."
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                    disabled={false}
                    id=""
                    name=""
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Selecione uma data..."
                    readOnly={true}
                    required={false}
                    type="text"
                    value="14 Mar 2019"
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <button
                  className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon icon-calendar Icon__StyledIcon-crr1cg-0 cmjVJQ"
                    disabled={false}
                    fontSize="1.75rem"
                    onClick={null}
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Horário de devolução
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormTime__StyledFormTime-sc-1ka3ing-1 bdjJPj"
        >
          <input
            className="FormTime__StyledDatePicker-sc-1ka3ing-0 hCpkqS"
            disabled={false}
            onChange={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  2,
                ],
                "delimiters": Array [
                  ":",
                ],
              }
            }
            placeholderText="Selecione um horário..."
            value="19:00"
          />
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Modelo do carro
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-carModel"
          disabled={false}
          id="carModel"
          maxLength={30}
          name="carModel"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="modelo do carro"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Passageiro responsável
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Documentos necessários
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormRadio__StyledOption-xdzjtw-1 dZGJdF"
        >
          <input
            checked={true}
            disabled={false}
            id="requiredDocuments-cnh"
            name="requiredDocuments-cnh"
            onChange={[Function]}
            type="radio"
            value="cnh"
          />
          <span
            className="FormRadio__StyledLabel-xdzjtw-0 cnAFWF"
          >
            CNH
          </span>
        </div>
        <div
          className="FormRadio__StyledOption-xdzjtw-1 dZGJdF"
        >
          <input
            checked={false}
            disabled={false}
            id="requiredDocuments-pid"
            name="requiredDocuments-pid"
            onChange={[Function]}
            type="radio"
            value="pid"
          />
          <span
            className="FormRadio__StyledLabel-xdzjtw-0 cnAFWF"
          >
            Permissão Internacional para Dirigir
          </span>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          O que inclui
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-library.isIncluded"
          disabled={false}
          id="isIncluded"
          maxLength={255}
          name="library.isIncluded"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
  </form>,
  <div
    className="FormButtonContainer-sc-1cqqivn-0 ADyLt"
  >
    <button
      className="Button__StyledButton-sc-1fc3klx-0 bEGBgE"
      disabled={false}
      fill={false}
      onClick={[Function]}
    >
      Salvar serviço
    </button>
    <a
      className="Cancel__StyledCancel-sc-1prgtit-0 WzuoR"
      to="/travel/5c52ebcd5fc7c949223ab705/script"
    >
      Cancelar
    </a>
  </div>,
]
`;
